server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -189,7 +189,7 @@ static SearchRequest subSearchRequest(
         return new SearchRequest(originalSearchRequest, indices, clusterAlias, absoluteStartMillis, finalReduce);
     }
 
-    private SearchRequest(
+    protected SearchRequest(
```

Comment:
We shouldn't relax `SearchRequest` visibility I believe, the way we approached similar usage is by using delegation